fluss-server/src/main/java/org/apache/fluss/server/zk/data/lake/LakeTable.java

```diff
@@ -0,0 +1,182 @@
+
+import org.apache.fluss.fs.FSDataInputStream;
+import org.apache.fluss.fs.FileSystem;
+import org.apache.fluss.fs.FsPath;
+import org.apache.fluss.server.zk.data.ZkData;
+import org.apache.fluss.utils.IOUtils;
+
+import javax.annotation.Nullable;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.fluss.metrics.registry.MetricRegistry.LOG;
+import static org.apache.fluss.utils.Preconditions.checkNotNull;
+
+/**
+ * Represents lake table snapshot information stored in {@link ZkData.LakeTableZNode}.
+ *
+ * <p>This class supports two storage formats:
+ *
+ * <ul>
+ *   <li>Version 1 (legacy): Contains the full {@link LakeTableSnapshot} data directly
+ *   <li>Version 2 (current): Contains a list of lake snapshot, recording the metadata file path for
+ *       different lake snapshots, with actual metadata storing in file to reduce zk pressure
+ * </ul>
+ *
+ * @see LakeTableJsonSerde for JSON serialization and deserialization
+ */
+public class LakeTable {
+
+    // Version 2 (current):
+    // a list of lake snapshot metadata, record the metadata for different lake snapshots
+    @Nullable private final List<LakeSnapshotMetadata> lakeSnapshotMetadata;
+
+    // Version 1 (legacy): the full lake table snapshot info stored in ZK, will be null in version2
+    @Nullable private final LakeTableSnapshot lakeTableSnapshot;
+
+    /**
+     * Creates a LakeTable from a LakeTableSnapshot (version 1 format).
+     *
+     * @param lakeTableSnapshot the snapshot data
+     */
+    public LakeTable(LakeTableSnapshot lakeTableSnapshot) {
+        this(lakeTableSnapshot, null);
+    }
+
+    /**
+     * Creates a LakeTable with a lake snapshot metadata (version 2 format).
+     *
+     * @param lakeSnapshotMetadata the metadata containing the file path to the snapshot data
+     */
+    public LakeTable(LakeSnapshotMetadata lakeSnapshotMetadata) {
+        this(null, Collections.singletonList(lakeSnapshotMetadata));
+    }
+
+    /**
+     * Creates a LakeTable with a list of lake snapshot metadata (version 2 format).
+     *
+     * @param lakeSnapshotMetadata the list of lake snapshot metadata
+     */
+    public LakeTable(List<LakeSnapshotMetadata> lakeSnapshotMetadata) {
+        this(null, lakeSnapshotMetadata);
+    }
+
+    private LakeTable(
+            @Nullable LakeTableSnapshot lakeTableSnapshot,
+            List<LakeSnapshotMetadata> lakeSnapshotMetadata) {
+        this.lakeTableSnapshot = lakeTableSnapshot;
+        this.lakeSnapshotMetadata = lakeSnapshotMetadata;
+    }
+
+    @Nullable
+    public LakeSnapshotMetadata getLakeTableLatestSnapshot() {
+        if (lakeSnapshotMetadata != null && !lakeSnapshotMetadata.isEmpty()) {
+            return lakeSnapshotMetadata.get(0);
+        }
+        return null;
+    }
+
+    @Nullable
+    public List<LakeSnapshotMetadata> getLakeSnapshotMetadata() {
```

Comment:
It returns a list of `LakeSnapshotMetadata`, better to name `getLakeSnapshotMetadatas`.